drivers/CoreLED/tests/CoreLED_test.cpp

```diff
@@ -0,0 +1,61 @@
+// Leka - LekaOS
+// Copyright 2022 APF France handicap
+// SPDX-License-Identifier: Apache-2.0
+
+#include "CoreLED.h"
+
+#include "gtest/gtest.h"
+#include "mocks/leka/SPI.h"
+
+using namespace leka;
+
+class CoreLEDTests : public ::testing::Test
+{
+  protected:
+	CoreLEDTests() : coreled(spimock) {}
+
+	// void SetUp() override {}
+	// void TearDown() override {}
+
+	CoreLED coreled;
+	SPIMock spimock;
+};
+
+TEST_F(CoreLEDTests, initialisation)
+{
+	ASSERT_NE(&coreled, nullptr);
+}
+
+TEST_F(CoreLEDTests, turnOff)
+{
+	coreled.turnOff();
+
+	ASSERT_EQ(false, coreled.isOn());
+}
+
+TEST_F(CoreLEDTests, turnOn)
+{
+	coreled.turnOn();
+
+	ASSERT_EQ(true, coreled.isOn());
+}
+
+TEST_F(CoreLEDTests, setImplementedColor)
+{
+	EXPECT_CALL(spimock, write(testing::ElementsAre(0x00, 0x00, 0x00, 0x00))).Times(2);
+	EXPECT_CALL(spimock, write(testing::ElementsAre(50, 0xFF, 0x00, 0x00))).Times(N_LEDS);
+	EXPECT_CALL(spimock, write(testing::ElementsAre(0x00, 0x00))).Times(1);
+	auto expected_belt_color = RGB::pure_red;
+	coreled.setColor(expected_belt_color);
+	coreled.showColor();
+}
```

Comment:
Super! C'est bien joué pour les `ElementsAre`!

Comme tu proposes d'envoyer plusieurs trames, j'imagine que l'ordre d'envoie à son importance. Si par exemple j'envoie `0x00,0x00,0x00,0x00` puis `0x00,0x00` puis `50, 0xFF, 0x00, 0x00` et enfin `0x00,0x00,0x00,0x00` est-ce que c'est juste? Les tests ici disent que oui

Il est possible d'imposer un ordre, avec `InSequence`, un autre outils de gtest.

```suggestion
using ::testing::InSequence;

...

TEST_F(CoreLEDTests, setImplementedColor)
{
	{
		InSequence seq;

		EXPECT_CALL(spimock, write(testing::ElementsAre(0x00, 0x00, 0x00, 0x00))).Times(1);
		EXPECT_CALL(spimock, write(testing::ElementsAre(50, 0xFF, 0x00, 0x00))).Times(N_LEDS);
		EXPECT_CALL(spimock, write(testing::ElementsAre(0x00, 0x00, 0x00, 0x00))).Times(1);
		EXPECT_CALL(spimock, write(testing::ElementsAre(0x00, 0x00))).Times(1);
	}

	auto expected_belt_color = RGB::pure_red;
	coreled.setColor(expected_belt_color);
	coreled.showColor();
}
```